### DISCUSSION TITLE
Using SharedLocalizer, what am I missing?

### DISCUSSION THREAD

**Discussion by Rodien** [oqtane/oqtane.framework#3351] [Q&A] [answered]
Quick question about localization:

I've added the following keys to 'SharedLocalizer', and I've also added these keys and their corresponding values to the .resx file.

![image](https://github.com/oqtane/oqtane.framework/assets/50739829/92ca0523-6d60-40c2-9d34-1b2a717ca0cb)

![image](https://github.com/oqtane/oqtane.framework/assets/50739829/70c70d73-ef87-478c-a542-a45913ca9d4f)

I am working in Oqtane.Client\Modules\Admin\Users\Index.razor, I added the values in \Oqtane.Client\Resources\Modules\Admin\Users\Index.resx

But on the UI I am seeing the keys and not the values.  So, I must be missing something? 

![image](https://github.com/oqtane/oqtane.framework/assets/50739829/7916efc3-6194-4b91-b63b-c4040ca39415)

@sbwalker @leigh-pointer any idea what I am missing?

**Comment by sbwalker** [comment, marked as answer]
@Rodien There are 2 types of RESX files which can be registered at the top of a razor component:

@inject IStringLocalizer< Edit > Localizer
@inject IStringLocalizer< SharedResources > SharedLocalizer

The Localizer refers to the RESX file corresponding to the current razor component. ie. Edit.razor = Edit.resx
The SharedLocalizer refers to a global file where "common" words or phrases are registered ( in Oqtane this is the Oqtane.Client\Resources\SharedResources.resx file )

So you should be using the Localizer - not the SharedLocalizer.

**Reply by Rodien** [reply]
Ah Got it! Thanks😁
